ppb_vector/vector2.py

```diff
@@ -34,6 +35,37 @@ def _mkvector(value, *, castto=_fakevector):
         raise ValueError(f"Cannot use {value} as a vector-like")
 
 
+@functools.lru_cache()
+def _find_lowest_type(left: type, right: type) -> type:
+    """
+    Guess which is the more specific type.
+    """
+    # Basically, see what classes are unique in each type's MRO and return who
+    # has the most.
+    lmro = set(left.__mro__)
+    rmro = set(right.__mro__)
+    lspecial = lmro - rmro
+    rspecial = rmro - lmro
+    if len(lmro) > len(rmro):
+        return left
+    elif len(rmro) > len(lmro):
+        return right
+    else:
+        # They're equal, just arbitrarily pick one
+        return left
+
+
+def _find_lowest_vector(left: type, right: type) -> type:
```

Comment:
Not actionable: I don't like the amount of private functions that this is growing into. I understand where it's coming from, thus, non-actionable.